api_app/serializers.py

```diff
@@ -592,3 +597,77 @@ class EnvelopeSerializer(rfs.Serializer):
         results = serializer_class(many=many)
 
     return EnvelopeSerializer
+
+
+class CustomConfigSerializer(rfs.ModelSerializer):
+    def validate_value(self, value):
+        try:
+            json.loads(value)
+        except json.JSONDecodeError:
+            raise ValidationError("Value is not JSON-compliant.")
+        return value
+
+    def validate(self, attrs):
+        super().validate(attrs)
+
+        # check if owner is admin of organization
+        if attrs.get("organization", None):
+            # check if the user is owner of the organization
+            membership = Membership.objects.filter(
+                user=attrs.get("owner"),
+                organization=attrs.get("organization"),
+                is_owner=True,
+            )
+            if not membership.exists():
+                raise ValidationError("User is not owner of the organization.")
+
+        config = None
+        category = None
+        if attrs["type"] == CustomConfig.Type.ANALYZER:
+            config = AnalyzerConfig
+            category = "Analyzer"
+        elif attrs["type"] == CustomConfig.Type.CONNECTOR:
+            config = ConnectorConfig
+            category = "Connector"
```

Comment:
cool optimization. could you also handle the impossible `else` case with a critical error log?